packages/client/src/components/app/forms/RelationshipField.svelte

```diff
@@ -1,39 +1,55 @@
-<script>
+<script lang="ts">
   import { CoreSelect, CoreMultiselect } from "@budibase/bbui"
   import { FieldType } from "@budibase/types"
   import { fetchData, Utils } from "@budibase/frontend-core"
   import { getContext } from "svelte"
   import Field from "./Field.svelte"
+  import type {
+    SearchFilter,
+    RelationshipFieldMetadata,
+    Table,
+    Row,
+  } from "@budibase/types"
 
   const { API } = getContext("sdk")
 
-  export let field
-  export let label
-  export let placeholder
-  export let disabled = false
-  export let readonly = false
-  export let validation
-  export let autocomplete = true
-  export let defaultValue
-  export let onChange
-  export let filter
-  export let datasourceType = "table"
-  export let primaryDisplay
-  export let span
-  export let helpText = null
-  export let type = FieldType.LINK
-
-  let fieldState
-  let fieldApi
-  let fieldSchema
-  let tableDefinition
-  let searchTerm
-  let open
+  export let field: string | undefined = undefined
+  export let label: string | undefined = undefined
+  export let placeholder: any = undefined
+  export let disabled: boolean = false
+  export let readonly: boolean = false
+  export let validation: any
+  export let autocomplete: boolean = true
+  export let defaultValue: string | undefined = undefined
+  export let onChange: any
+  export let filter: SearchFilter[]
+  // not really obvious how to type this - some components pass other things here
+  // but it looks like the component data fetch should only work with tables
+  export let datasourceType: "table" = "table"
```

Comment:
I think this was added to support the other BB reference types. For example in `BBReferenceField` we pass in `datasourceType="user"`, and looking at the possible types I think this could either be `table`, `user` or `groupUser`.

Reply:
So this causes type errors on line 37 with:
```
  $: fetch = fetchData({
    API,
    datasource: {
      type: datasourceType,
      tableId: linkedTableId,
    },
    options: {
      filter,
      limit: 100,
    },
  })
```

This seems to think the only valid `datasourceType` is `"table"` - I didn't want to fiddle with the typing of data fetching, but something doesn't seem quite right there.